consensus/consensus_v2.go

```diff
@@ -119,88 +181,81 @@ func (consensus *Consensus) finalizeCommits() {
 	consensus.aggregatedCommitSig = aggSig // this may not needed
 	consensus.FBFTLog.AddMessage(FBFTMsg)
 	// find correct block content
-	curBlockHash := consensus.blockHash
-	block := consensus.FBFTLog.GetBlockByHash(curBlockHash)
+
+	block := consensus.FBFTLog.GetBlockByHash(consensus.BlockHash())
 	if block == nil {
-		consensus.getLogger().Warn().
-			Str("blockHash", hex.EncodeToString(curBlockHash[:])).
-			Msg("[FinalizeCommits] Cannot find block by hash")
-		return
+		utils.Logger().Warn().Msg("[FinalizeCommits] Cannot find block by hash")
+		return errors.New("could not find block by hash")
 	}
 
-	consensus.tryCatchup()
-	if consensus.blockNum-beforeCatchupNum != 1 {
-		consensus.getLogger().Warn().
-			Uint64("beforeCatchupBlockNum", beforeCatchupNum).
-			Msg("[FinalizeCommits] Leader cannot provide the correct block for committed message")
-		return
+	if err := consensus.tryCatchup(); err != nil {
+		return err
 	}
 
+	// if consensus.BlockNum()-beforeCatchupNum != 1 {
+	// 	utils.Logger().Warn().
+	// 		Uint64("beforeCatchupBlockNum", beforeCatchupNum).
+	// 		Msg("[FinalizeCommits] Leader cannot provide the correct block for committed message")
+	// 	return errors.New("leader cannot provide the correct block for committed message")
+	// }
+
 	// if leader success finalize the block, send committed message to validators
-	if err := consensus.msgSender.SendWithRetry(
```

Comment:
why remove the retry logic? I think the retry logic can same us from potential failures, even when the p2p layer is stronger than before, problems may still happen.

Reply:
it will/can come back - but its has its own mutexes that lock up - first need correctness, then easy to add retry on top if needed -